app/Http/Controllers/admin/TopicController.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace App\Http\Controllers\admin;
+
+use App\Http\Requests\TopicFormRequest;
+use App\Models\Topic;
+use App\Repositories\Contracts\TopicRepository;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Facades\Input;
+use Illuminate\Support\Facades\Redirect;
+use Illuminate\Support\Facades\Storage;
+
+class TopicController extends Controller
+{
+
+    protected $topic;
+
+    public function __construct(TopicRepository $topic)
+    {
+        $this->topic = $topic;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $topics = $this->topic->getAll();
+        $test_model = new Topic();
+        $fillable_columns = $test_model->getFillable();
+        foreach ($fillable_columns as $key => $value) {
+            $test_columns[$value] = $value;
+        }
+
+        return view('admin.topic.index', compact('topics', 'test_columns'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('admin.topic.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(TopicFormRequest $request)
+    {
+        if (Auth::check()) {
+            $user_id = Auth::user()->id;
+            $request->merge(['user_id' => $user_id]);
+            $image = $request->file('image_topic');
+            $img = $this->saveImage($image);
+            $request->merge(['images' => $img]);
+
+            //handle select time
+            if ($request->input('set_time') == 0) {
+                $select_time = null;
+            } elseif ($request->input('set_time') == 1) {
+                $select_time = $request->input('select_time');
+
+            }
+            $request->merge(['select_time' => $select_time]);
+//            dd(Carbon::now()->hour . ':'. Carbon::now()->minute);
+            $this->topic->store($request->all());
+
+            return back()->with('success', __('message.success'));
+        }
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionClosingBrace.SpacingBeforeClose</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 83</br>Function closing brace must go on the next line following the body; found 1 blank lines before brace